wgsl/index.bs

```diff
@@ -9398,12 +9432,18 @@ Each user-defined [=user-defined input datum|input=] and [=user-defined output d
 Each structure member in the entry point IO [=shader-creation error|must=] be one of either a built-in value
 (see [[#builtin-inputs-outputs]]), or assigned a location.
 
-Locations [=shader-creation error|must not=] overlap within each of the following sets:
+
+For each set of outputs |outputs| in the following sets:
 * Members within a structure type.
     This applies to any structure, not just those used in shader stage inputs or outputs.
 * An entry point's shader stage inputs,
     i.e. locations for its formal parameters, or for the members of its formal parameters of structure type.
 
+The following [=shader-creation error|must=] be `true`:
+- |outputs| [=shader-creation error|must not=] contain two outputs with the same location and without a given [=attribute/blend_src=] attribute.
+- If any output in |outputs| specifies a [=attribute/blend_src=] attribute:
+    |outputs| must contain exactly 2 outputs, one with `@location(0) @blend_src(0)` and one with `@location(0) @blend_src(1)`.
```

Comment:
`algorithm` is required to make variables work nicely. Also it's clearer to associate these two paragraphs together.

However, apologies, I'm rereading this (and the text I suggested) and realizing that I totally misunderstood what the second bullet is about - it's about inputs, not outputs. I guess just looking at structs is good enough for outputs (since a struct is the only way to have more than one output of an entry point, currently, I think).

```suggestion
<div algorithm="locations in entry point inputs">
    For each entry point defined in a WGSL module, let |inputs| be its set of shader stage inputs
    (i.e. locations for its formal parameters, or for the members of its formal parameters of structure type).
    
    - |inputs| [=shader-creation error|must not=] contain two entries with the same [=attribute/location=] value.
</div>

<div algorithm="locations in structs">
    For each structure type defined in a WGSL module (not just those used in shader stage inputs or outputs),
    let |members| be the set of struct members with [=attribute/location=] attributes.

    - |members| [=shader-creation error|must not=] contain two entries without [=attribute/blend_src=]
        attributes that have the same with the same [=attribute/location=] value.
    - If any entry in |members| specifies a [=attribute/blend_src=] attribute:
        - |members| [=shader-creation error|must=] contain exactly 2 entries,
            one with `@location(0) @blend_src(0)` and one with `@location(0) @blend_src(1)`.
</div>
```

@alan-baker WDYT, does this seem correct?

Reply:
Done